puppet/environments/box/modules/profiles/files/box/cli/scripts/Command/Share.php

```diff
@@ -0,0 +1,144 @@
+<?php
+namespace Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Share extends Command
+{
+    protected $site;
+    protected $www;
+    protected $vhost_dir;
+    protected $vhost_file;
+    protected $tmp_dir;
+
+    protected function configure()
+    {
+        $this->setName('share')
+             ->setDescription('Share a local site with a colleague')
+             ->setHelp(<<<EOF
+This command will start an Ngrok (https://ngrok.com/) session which will give you a URL to share with your colleagues. 
+When you're finished, press `ctrl`+`c` to stop ngrok and close the session.
+
+<info>Note:</info> you need to SSH into the box with `vagrant ssh` first to run this command. This command does not work in the web terminal!
+EOF
+            )
+            ->addArgument(
+                'site',
+                InputArgument::OPTIONAL,
+            'Provide the site name of the website you wish to share',
+                ''
+            )
+            ->addOption(
+                'www',
+                'd',
+                InputOption::VALUE_REQUIRED,
+                "Web server root",
+                '/var/www'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->vhost_dir = '/etc/apache2/sites-available';
+        $this->vhost_file = '2-ngrok.conf';
+        $this->tmp_dir = '/tmp';
+
+        $this->www = $input->getOption('www');
+        $this->site = $input->getArgument('site');
+
```

Comment:
@stevenrombauts I'm just wondering if we need to clear out any previous screens before attempting to launch a new share? I've found a few times `box share leadman` just leads to empty output and no share. I'm guessing this might be because a share might have already been started and not shut down properly. Restarting the box fixes this... but I've just found: 

`killall screen` 

Which I think might solve the situation for itself.

Reply:
@yiendos That makes sense. We should give that a try. Would there be any way to be more specific about which process we kill (instead of just all instancnes of screen)?

Reply:
@stevenrombauts opening a share with a `share_name` seems to work quite well:
```
//launch ngrok and create a new screen to keep the user on the foreground
        `screen -S ngrokShare -d -m $ngrok_command`;
```

Which then lists as:
```
$ screen -ls
There is a screen on:
	8007.ngrokShare	(06/27/2019 01:21:25 PM)	(Attached)
1 Socket in /run/screen/S-vagrant.
```

And can easily be killed by:
```
 screen -S ngrokShare -X quit
```

So we can be a lot more targeted in our approach. Should I open another ticket to handle this?

Reply:
@yiendos Excellent find, that looks like it could be a great solution. Let's give it a go in another ticket then.